packages/cloud-cognitive/src/components/CreateTearsheet/CreateTearsheetStep.js

```diff
@@ -22,68 +22,93 @@ export let CreateTearsheetStep = forwardRef(
     {
       children,
       className,
-      subtitle,
       description,
       disableSubmit,
-      introStep,
-      title,
-      hasFieldset,
       fieldsetLegendText,
+      hasFieldset,
+      includeStep,
+      introStep,
       onNext,
       onMount,
       secondaryLabel,
+      subtitle,
+      title,
     },
     ref
   ) => {
-    const steps = useContext(StepsContext);
+    const stepsContext = useContext(StepsContext);
     const stepNumber = useContext(StepNumberContext);
-    const previousState = usePreviousValue({ currentStep: steps?.currentStep });
+    const [shouldIncludeStep, setShouldIncludeStep] = useState();
+    const previousState = usePreviousValue({
+      currentStep: stepsContext?.currentStep,
+    });
 
     // This useEffect reports back the onNext and onMount values so that they can be used
     // in the appropriate custom hooks.
     useEffect(() => {
       if (
-        stepNumber === steps?.currentStep &&
-        previousState?.currentStep !== steps?.currentStep
+        stepNumber === stepsContext?.currentStep &&
+        previousState?.currentStep !== stepsContext?.currentStep
       ) {
-        steps?.setOnNext(onNext);
-        steps?.setOnMount(onMount);
+        stepsContext?.setOnNext(onNext);
+        stepsContext?.setOnMount(onMount);
       }
-    }, [onMount, onNext, steps, stepNumber, previousState?.currentStep]);
+    }, [onMount, onNext, stepsContext, stepNumber, previousState?.currentStep]);
+
+    // Used to take the `includeStep` prop and use it as a local state value
+    useEffect(() => {
+      setShouldIncludeStep(includeStep);
+    }, [includeStep, stepsContext, title]);
 
     // This useEffect makes sure that every CreateTearsheetStep reports back it's
     // title, secondaryLabel, and introStep props so that it can be sent to the CreateInfluencer.
     useEffect(() => {
-      const stepHasReported = steps?.stepData?.includes(
-        (item) => item.title === title
-      );
-      if (!stepHasReported && steps?.stepData?.length < steps?.totalStepCount) {
-        steps?.setStepData((prev) => [
-          ...prev,
-          {
+      if (stepsContext) {
```

Comment:
same comment about reusing code

Reply:
Updated, this part here as well as the same part in `CreateFullPageStep` are now using a custom hook called `useRetrieveStepData`.